### PR TITLE
feat: sort-order rule shouldn't change sub-keys, better auto-fix

### LINKED ISSUE #80: 🐛 Bug: package-json/order-properties changes scripts keys
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

It should only sort the top-level keys

### Actual

It complains about key order in deeper paths as well

### Additional Info

With this config
```json
{
  "files": ["package.json"],
  "plugins": ["package-json"],
  "rules": {
    "package-json/order-properties": [
      "error",
      {
        "order": [
          "name",
          "scripts",
          "dependencies"
        ]
      }
    ],
    "package-json/sort-collections": [
      "error", 
      ["dependencies", "devDependencies", "peerDependencies"]
    ]
  }
}
```

I'm expecting this
```json
{
  "scripts": {
    "start": "",
    "dev": ""
  }
  "name": "",
  "dependencies": []
}
```

To only reorder on top-level, like this
```json
{
  "name": "",
  "scripts": {
    "start": "",
    "dev": ""
  },
  "dependencies": []
}
```

But it seems to change scripts order as well
```json
{
  "name": "",
  "scripts": {
    "dev": "",
    "start": ""
  },
  "dependencies": []
}
```

I'm stuck on `^0.1.5` because of this :disappointed:

### PR DESCRIPTION
**Pull request by hyoban**
## PR Checklist

-   [x] Addresses an existing open issue: fixes #80 
-   [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/blob/main/.github/CONTRIBUTING.md) were taken

### CODE REVIEW

**Comment by hyoban** [comment]
@JoshuaKGoldberg Sorry for the direct ping, I don't want you to miss this one. Please feel free to review it when you have time.

File: src/tests/rules/order-properties.test.ts
Comment (JoshuaKGoldberg) [inline src/tests/rules/order-properties.test.ts:1]: [Testing] Could you add a `valid` case where the top-level keys are sorted but sub-keys aren't? That way we know the logic is still just looking at the top-level keys.

File: src/rules/order-properties.ts [hunk 1]
```diff
@@ -73,10 +76,24 @@ export const rule = createRule<Options>({
...
+									) + endCharacters;
+								if (newline === "\r\n") {
+									result = result.replace(/\n/g, newline);
+								}
+
 								return fixer.replaceText(
 									context.sourceCode.ast,
-									JSON.stringify(orderedSource, null, 2) +
-										`\n`,
+									result,
```
Comment (JoshuaKGoldberg) [inline src/rules/order-properties.ts:96]: [Praise] Nice, this results in better-preserved formatting for all auto-fixes!

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
This looks great, thanks! Just the one testing request. 🚀 

Also - I think this is a big enough change (new dependencies, better auto-fix) that even though it's technically a bugfix, I'd like it to give it as a minor release rather than a patch. Renaming to `feat: ...`.

**Comment by JoshuaKGoldberg** [comment]
Thanks for the ping! I've been in and out of being swamped with ESLint v9 / typescript-eslint v8 work, just emerging from that cocoon now. 😄

**Review by JoshuaKGoldberg — APPROVED** [review]
A bunch of triumphant Mets players cheering while one guy in particular is whooping and walking

**Pull request merged**